python/pyarrow/tests/test_compute.py

```diff
@@ -230,6 +230,51 @@ def test_match_substring():
     assert expected.equals(result)
 
 
+def test_split_pattern():
```

Comment:
It doesn't really make sense to duplicate C++ tests on the Python side.

Reply:
It is more a coverage/sanity check, it is not a duplication of the C++ test. I'd like to keep it for now, as it triggered many issues during dev/refactoring.